File: frontend/src/app/model/proveedor.ts
export interface Proveedor {
  id?: string;
  nombre: string;
  nif?: string;
  email?: string;
  telefono?: string;
  direccion?: string;
  contacto?: string;
  notas?: string;
  proveedorMaterial?: boolean;
  proveedorReparacion?: boolean;
}

File: backend/app/src/main/java/com/david/app/model/Proveedor.java
package com.david.app.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "proveedores")
@Data
public class Proveedor {
    @Id
    private String id;

    private String nombre;
    private String nif;
    private String email;
    private String telefono;
    private String direccion;
    private String contacto;
    private String notas;
    private Boolean proveedorMaterial;
    private Boolean proveedorReparacion;
}
